search/geocoder.cpp

```diff
@@ -1121,26 +1125,26 @@ void Geocoder::MatchPOIsAndBuildings(BaseContext & ctx, size_t curToken)
       }
 
       layer.m_type = static_cast<SearchModel::SearchType>(i);
-      if (IsLayerSequenceSane())
+      if (IsLayerSequenceSane(layers))
         MatchPOIsAndBuildings(ctx, curToken + n);
     }
   }
 }
 
-bool Geocoder::IsLayerSequenceSane() const
+bool Geocoder::IsLayerSequenceSane(vector<FeaturesLayer> const & layers) const
 {
-  ASSERT(!m_layers.empty(), ());
+  ASSERT(!layers.empty(), ());
   static_assert(SearchModel::SEARCH_TYPE_COUNT <= 32,
                 "Select a wider type to represent search types mask.");
   uint32_t mask = 0;
-  size_t buildingIndex = m_layers.size();
-  size_t streetIndex = m_layers.size();
+  size_t buildingIndex = layers.size();
+  size_t streetIndex = layers.size();
```

Comment:
No thanks. Because all other code in this file does not follow this "consts, consts for everything!" phylosophy.